### ISSUE TITLE
 Problem with integration

### ISSUE THREAD

**Issue by ripolltata** [IDmedia/hass-nohassle_hdmi_matrix#1]
Hello good Morning. I am fairly new to home assistant, I purchased a no-hassle 8x8 matrix and installed the repository by hacs, then add the example to the configuration.yaml
Now I have the media player to add in lovelace but it always appears off and I can't find how to see the state of my matrix and I can't change the input as indicated below.
Could you give an example of how to do to change the source and what else would you need to put in addition to the example of the configuration.yaml?

**Comment by IDmedia** [comment]
You have it connected by Ethernet and remembered to change the host to match yours?
I use this integration every day without any issues, but they may have changed something in newer revisions of the Matrix.. mine is a few years older model.

**Comment by Absinthze** [comment]
I am the same as ripolltata above.  Very new, and coding is a new thing to me.  Can you please explain how to make a card that will have switches similar to the android app.   
Do I have to make a script first?   
Thank you!

**Comment by IDmedia** [comment]
@Absinthze Please create a new issue instead of hijacking this one with a different problem.

With that said, as long as triggering `media_player.hdmi_matrix_set_zone` works (see 'Usage') by visiting 'Developer Tools->Services' then please ask in the Home Assistant forums for specific help on how to design your UI. I personally use the [button card](https://github.com/custom-cards/button-card)  module and have never used the Android app.

**Comment by Absinthze** [comment]
> 
> 
> @Absinthze Please create a new issue instead of hijacking this one with a different problem.
> 
> With that said, as long as triggering `media_player.hdmi_matrix_set_zone` works (see 'Usage') by visiting 'Developer Tools->Services' then please ask in the Home Assistant forums for specific help on how to design your UI. I personally use the [button card](https://github.com/custom-cards/button-card) module and have never used the Android app.

Sorry, I've never posted on Github before, dont know the rules.   I tried testing changing the zones, in Services with this line as you described.  But nothing happens.   I have the config YAML entered with these descriptions.  Is there a single line I'm missing?

service: media_player.hdmi_matrix_set_zone
data:
  entity_id: media_player.matt_office
  source: Chrome

**Comment by IDmedia** [comment]
I may see the problem. After you mentioned the Android app I installed it and I see it does not work with my HDMI Matrix, even though it says it's connected. I may be on an older revision/version of the HDMI Matrix hardware or firmware. I'll look into the Android app and figure out what they do when you switch sources and if that doesn't yield any results I'll get in touch with NoHassle again and ask them for a clarification. Please give me a few days to look into this.

**Comment by IDmedia** [comment]
I have not yet heard anything from NoHassle and I'm a bit dissapointed to be honest. I would love to fix this issue.
However, maybe one of you can help me sort this out by doing the following:

1. Use Chrome to navigate to your Matrix's WebUI. For example: http://192.168.1.168/
2. Hit "F12" in order to open Chrome Developer Tools
3. Click on "Network" and you should see all requests that page generates
4. Hit "Input 1" button for all sources and make sure the Matrix switches source. The request should be visible in the Network-tab in the Chrome Developer Tools
5. Right click on ANY of the rows and select "Save all as HAR with content" and save the .har file somewhere
6. Upload it here, it does not contain any personal information as long as you don't login to your WebUI (which I don't think it supports/mine doesn't).

My HDMI Matrix has requests such as this one when selecting Input1 for Output 1:
`http://192.168.1.168/@PORT1=1.0.8801356435426098`

I also see this one when the WebUI updates which I use to determine the currently active source:
`http://192.168.1.168/AutoGetAllData=0.13093804061090375`

So my hopes is that you guys have similar requests, but differ maybe a little bit since you have later revisions....

**Comment by Absinthze** [comment]
Hi Benjamin
I saved the HAR file (attached), and took a screen capture of the developer
page.    However I don't see any of the IP address you mentioned.  Let me
know how else I can help.
[image: image.png]

**Comment by IDmedia** [comment]
@Absinthze  That didn't seem to work. I can't see the screenshot or the attachment.

**Comment by IDmedia** [comment]
Absinthze was really helpful and I got all info I need for now. I'll try to add support for your revision of HDMI Matrix in the next couple of days. I may take some time though as I'm very busy with other projects right now. I'll update this issue once a beta-version is ready :)

**Comment by IDmedia** [comment]
Please test the latest version and let me know how it works :)

**Comment by IDmedia** [comment]
The latest version should work now. Please open a new issue if necessary :)
